### PR TITLE
SW-4617 Added useQueryFilter hook that syncs between a consumer's filters and the query string in the address bar

### PR DESCRIPTION
**Pull request by nickgraz**
SW-4617 Added useQueryFilter hook that syncs between a consumer's filters and the query string in the address bar

### CODE REVIEW

**Comment by nickgraz** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #1965** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/1965?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #1966** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/1966?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #1967** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/1967?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/1965?utm_source=stack-comment).

File: src/utils/useQueryFilters.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      .join(',');
+  } else if (value === null) {
+    return '';
+  } else {
+    return `${value}`;
+  }
+};
+
+const denormalizeNumber = (value: string): number | string =>
+  value.match(/^[0-9]+$/) && `${Number(value)}` === value ? _.toFinite(value) : value;
```
Comment (karthikbtf) [inline src/utils/useQueryFilters.ts:27]: do we care about floating point?

**Review by karthikbtf — APPROVED** [review]

File: src/utils/useQueryFilters.ts (hunk 1 above)
Comment (nickgraz) [reply src/utils/useQueryFilters.ts:27]: We may in the future, but as far as I can tell we only have filters that are either a boolean or an array of numbers

**Comment by nickgraz** [comment]
### Merge activity

* **Jan 3, 3:04 PM**: @nickgraz started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/1965).

**Pull request merged**